mvlearn/decomposition/baseica.py

```diff
@@ -0,0 +1,233 @@
+# MIT License
+
+# Copyright (c) [2017] [Pierre Ablin and Hugo Richard]
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import numpy as np
+from scipy import linalg
+from sklearn.decomposition import PCA
+
+from .base import BaseDecomposer
+from ..preprocessing.repeat import ViewTransformer
+
+
+class BaseICA(BaseDecomposer):
+    """A base class for multiview ICA methods.
+
+    Parameters
+    ----------
+    n_components : int, optional
+        Number of components to extract. If None, n_components is set to
+        the minimum number of features in the dataset.
+
+    preproc: None, 'pca' or a ViewTransformer-like instance,\
```

Comment:
would a pipeline be easier? rather than making the model accept an estimator as init param?

Reply:
I think multiview ICA should come with a standard groupICA initialization, as it should be the preferred init in 90% of the cases. 

And then if the user wants to do something more complicated he should rather use a pipeline.

Reply:
The current implementation allows the user to do n_components = 5 and do more or less what Pierre suggests without worrying about anything else.
Preproc is only to be used by advanced users and is indeed equivalent to a pipeline.